repos/system_upgrade/el7toel8/actors/unsupportedupgradecheck/actor.py

```diff
@@ -0,0 +1,66 @@
+import os
+import json
+
+from leapp.actors import Actor
+from leapp.models import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.utils.audit import get_connection
+from leapp import reporting
+
+
+class UnsupportedUpgradeCheck(Actor):
+    """
+    Checks enviroment variables and produces a warning report if the upgrade is unsupported.
+
+    Upgrade is unsupported if any LEAPP_DEVEL_* variable is used or an experimental actor is enabled.
+    This can be overridden by setting the variable LEAPP_UNSUPPORTED (at user's own risk).
+    """
+
+    name = 'unsupported_upgrade_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        leapp_vars = self.configuration.leapp_env_vars
+        devel_vars = [var for var in leapp_vars if var.name.startswith('LEAPP_DEVEL_')]
+        override = bool([var for var in leapp_vars if var.name == 'LEAPP_UNSUPPORTED'])
+
+        if devel_vars and not override:
+            reporting.create_report([
+                reporting.Title('Upgrade inhibited due to usage of development variables'),
+                reporting.Summary(
+                    'One or more environment variables in the form of LEAPP_DEVEL_* have been detected.\n'
+                    'These variables are for development purposes only and can interfere with the upgrade '
+                    'process in unexpected ways. As such, a successful and safe upgrade process cannot be '
+                    'guaranteed and the upgrade is unsupported.\n'
+                    'Found development variables:\n- ' + '\n- '.join([var.name for var in devel_vars]) + '\n'
+                ),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Flags([reporting.Flags.INHIBITOR]),
+                reporting.Tags([reporting.Tags.UPGRADE_PROCESS, reporting.Tags.SANITY]),
+                reporting.Remediation(hint='Invoke leapp without any LEAPP_DEVEL_* environment variables.')
+            ])
+
+        experimental = []
+        with get_connection(None) as db:
+            conf = db.execute("SELECT configuration FROM execution "
+                              "WHERE kind = 'upgrade' OR kind = 'preupgrade' "
```

Comment:
Why are we selecting kind=upgrade? Do we have post-upgrade reporting as well already?

Reply:
And another stupid question - shouldn't we be querying for some specific execution id instead of select * limit 1?

Reply:
@fernflower I would rather ask, why there is communication with db itself from the actor side. That's playground of framework and actors should receive the information in different way.

Reply:
@pirat89 @fernflower alright. How can I get this information without querying the database? As far as I can see, information about experimental actors is not in the configuration.

Reply:
@fernflower 
>shouldn't we be querying for some specific execution id instead of select * limit 1?

Yeah, that would be better.

Reply:
No queries now!